lib/ecto/query.ex

```diff
@@ -438,12 +443,29 @@ defmodule Ecto.Query do
       end)
 
   """
-  def subquery(%Ecto.SubQuery{} = subquery),
-    do: subquery
-  def subquery(%Ecto.Query{} = query),
-    do: %Ecto.SubQuery{query: query}
-  def subquery(queryable),
-    do: %Ecto.SubQuery{query: Ecto.Queryable.to_query(queryable)}
+  def subquery(query_source, opts \\ []) do
+    query_source = if Keyword.has_key?(opts, :prefix) do
```

Comment:
Thank you! I think this code could be slightly cleaner though. Something like this:

```elixir
def subquery(query, opts \\ []) do
  subquery = wrap_query(query)
  case Keyword.fetch(opts, :prefix) do
    {:ok, prefix} when is_binary(prefix) -> put_in subquery.query.prefix, prefix
    :error -> subquery
  end
end
```

Reply:
@josevalim Updated per your view and pushed. At one point I did originally have the `Keyword.fetch` style, but the `put_in` part was great! Learning for me :)